app/parse.py

```diff
@@ -48,6 +52,67 @@ def parse_marc(content: bytes) -> list[Incomplete]:
     return entries
 
 
+def parse_excel(content: bytes) -> list[Incomplete]:
+    entries: list[Incomplete] = []
+    # Create a file-like object from the decoded bytes
+    file_like: BytesIO = io.BytesIO(content)
+    sheet = pandas.read_excel(file_like)
+    # Replace bogus values with empty strings
+    sheet.fillna("", inplace=True)
+    # Normalize column names once
+    sheet.rename(
+        columns=lambda x: x.strip().replace(" ", "_").replace("/", "_"), inplace=True
+    )
+    # Detect problematic column names
+    title_column = next((col for col in sheet.columns if col.startswith("Title")), None)
+    reading_level_column = next(
+        (col for col in sheet.columns if col in ["Lexile", "Reading_Level"]), ""
+    )
+
+    for row in sheet.itertuples(index=False):
```

Comment:
Highly recommend looking into [Dataframe iterrows()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.iterrows.html#pandas-dataframe-iterrows), which will probably remove the use of if else here

Reply:
Actually I **started** with that one particularly because it was a bit easier to read/follow, but it was extremely non-performant. My file with 80,000 rows took way longer than a full minute or two. I switched to `itertuples` and the same file parsing now takes around 30~ seconds.

Reply:
interesting. I wouldn't recommend `pandas` for these types of tasks (parsing the file to only get data) since `pandas` is more for data science stuff. But for now this works